LCM/dsc/engine/dsc_host/dsc_host.c

```diff
@@ -0,0 +1,339 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <locale.h>
+
+#include "dsc_sal.h"
+#include "DSC_Systemcalls.h"
+#include "Resources_LCM.h"
+#include "EngineHelper.h"
+#include "EventWrapper.h"
+#include "dsc_host.h"
+#include "dsc_library.h"
+#include "lcm/strings.h"
+
+#if defined(BUILD_OMS)
+extern MI_Boolean g_DscHost;
+#endif
+
+MI_Result GetDSCHostVersion(_In_z_ MI_Char* version, _In_ size_t length)
+{
+    if(Stprintf(version, length, MI_T("Version : %d.%d.%04d"), MajorVersion, MinorVersion, BuildVersion) < 0)
+        return MI_RESULT_FAILED;
+    else
+        return MI_RESULT_OK;
+}
+
+void PrintVersion()
+{
+    MI_Char versionInfo[DSCHOST_VERSION_BUFF_SIZE];
+    GetDSCHostVersion(versionInfo, DSCHOST_VERSION_BUFF_SIZE);
+    Tprintf(versionInfo);
+}
+
+void PrintHelp()
+{
+    Tprintf(MI_T("Usage:\n"));
+    Tprintf(MI_T("dsc_host [--help] [--version]\n"));
+    Tprintf(MI_T("dsc_host <Output Folder Path> <Operation> [Operation Arguments] \n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Supported Operation values are:\n"));
+    Tprintf(MI_T("  SendConfiguration <MOF Document Path>\n"));
+    Tprintf(MI_T("  SendConfigurationApply <MOF Document Path>\n"));
+    Tprintf(MI_T("  GetConfiguration [Configuration Document Path]\n"));
+    Tprintf(MI_T("  ApplyConfiguration\n"));
+    Tprintf(MI_T("  SendMetaConfigurationApply <MOF Document Path> [force]\n"));
+    Tprintf(MI_T("  GetMetaConfiguration\n"));
+    Tprintf(MI_T("  RollBack\n"));
+    Tprintf(MI_T("  TestConfiguration\n"));
+    Tprintf(MI_T("  PerformRequiredConfigurationChecks [flags value]\n"));
+    Tprintf(MI_T("  StopConfiguration [force]\n"));
+    Tprintf(MI_T("  PerformInventory\n"));
+    Tprintf(MI_T("  PerformInventoryOOB [MOF Document Path]\n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Examples:\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfiguration ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfigurationApply ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetConfiguration ./GetAuditPolicy.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output ApplyConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendMetaConfigurationApply ./new_metaconfig.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetMetaConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output RollBack\n"));
```

Comment:
we dont support rollback on linux/windows. later user may use host directly. i think we should not expose this functionality from host. or keep it hidden, do not show in help

Reply:
The scripts already exist and call into engine through omi. This is just there to keep all operations as before.

---
In reply to: [276395186](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/553#discussion_r276395186) [](ancestors = 276395186)

Reply:
then hide the functionality. dont show this in help?

---
In reply to: [276415215](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/553#discussion_r276415215) [](ancestors = 276415215,276395186)